pkg/apis/pipeline/v1alpha1/pipelinerun_types.go

```diff
@@ -208,3 +211,27 @@ func (pr *PipelineRun) GetServiceAccountName(pipelineTaskName string) string {
 	}
 	return serviceAccountName
 }
+
+// PipelineRunTaskSpec  can be used to configure specific
+// specs for a concrete Task
+type PipelineRunTaskSpec struct {
+	TaskName               string       `json:"taskName,omitempty"`
+	TaskServiceAccountName string       `json:"serviceAccountName,omitempty"`
+	TaskPodTemplate        *PodTemplate `json:"podTemplate,omitempty"`
+}
+
+// GetTaskSpec returns the task specific spec for a given
+// PipelineTask if configured, otherwise it returns the PipelineRun's default.
+func (pr *PipelineRun) GetTaskSpec(pipelineTaskName string) (string, *PodTemplate) {
+	// To make it backward compatible, until ServiceAccountNames is dispatched.
+	// serviceAccountName := pr.Spec.ServiceAccountName
```

Comment:
Remove?

Reply:
If we add the field, we have to support both and decide of a "precedence" between `serviceAccountNames` and this one.

My vote would be to prefer this over `serviceAccountNames` (if this field is specified)

Reply:
But would we like to keep the the [serviceAccountName](https://github.com/tektoncd/pipeline/blob/7cb3f4888c4a78b6ab4d7bdc6795be44303dce61/pkg/apis/pipeline/v1alpha1/pipelinerun_types.go#L78)  which then will be used for all task steps similar to how the [podTemplate](https://github.com/tektoncd/pipeline/blob/7cb3f4888c4a78b6ab4d7bdc6795be44303dce61/pkg/apis/pipeline/v1alpha1/pipelinerun_types.go#L89) or only have the avialbility to set it on task level? I suggest that we remove the serviceAccountNames, but not the serviceAccountName. 

When serviceAccountNames potential is removed we could use the ```serviceAccountName := pr.Spec.ServiceAccountName``` as the default for each step?

Reply:
So `pr.GetServiceAccountName(pipelineTaskName)` works just fine (aka as of today), we can remove the comments then